sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/ConnectedApDTO.java

```diff
@@ -53,4 +55,16 @@ public static ConnectedApDTO from(ConnectedAPAdapter connectedAPAdapter) {
 
         return connectedApDTO;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
ConnectedApDTO could be a record.
If it is too much work in this issue, you can simply use `@EqualsAndHashCode` lombok annotation instead of implementing equals and hascode.

Record proposal : 
```java
// SPDX-FileCopyrightText: 2021 RTE FRANCE
//
// SPDX-License-Identifier: Apache-2.0

package org.lfenergy.compas.sct.commons.dto;

import org.lfenergy.compas.sct.commons.scl.com.ConnectedAPAdapter;

/**
 * A representation of the model object <em><b>Connected AP</b></em>.
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link ConnectedApDTO#apName() <em>Ap Name</em>}</li>
 *   <li>{@link ConnectedApDTO#iedName() <em>Ied Name</em>}</li>
 * </ul>
 *
 * @see org.lfenergy.compas.scl2007b4.model.TConnectedAP
 */
public record ConnectedApDTO(String iedName, String apName) {

    /**
     * Convert ConnectedAPAdapter object to dto ConnectedApDTO
     *
     * @param connectedAPAdapter object to convert
     * @return dto ConnectedApDTO
     */
    public static ConnectedApDTO from(ConnectedAPAdapter connectedAPAdapter) {
        return new ConnectedApDTO(connectedAPAdapter.getIedName(), connectedAPAdapter.getApName());
    }

}
```

Reply:
ok